### PR TITLE
Improve doc for Kernel.SpecialForms.&/1

### PR DESCRIPTION
**Pull request by taiansu**
* Add the example of partially applied module function
* Improve local function example
* Improve list example

### CODE REVIEW

File: lib/elixir/lib/kernel/special_forms.ex [hunk 1]
```diff
@@ -1437,11 +1437,18 @@ defmodule Kernel.SpecialForms do
       4
 
   In other words, `&(&1 * 2)` is equivalent to `fn x -> x * 2 end`.
-  Another example using a local function:
 
-      iex> fun = &is_atom(&1)
-      iex> fun.(:atom)
-      true
+  We can pritially apply a module function with placeholder.
```
Comment (josevalim) [inline lib/elixir/lib/kernel/special_forms.ex]: "pritially" => "partially".

"module function" => "remote function".

File: lib/elixir/lib/kernel/special_forms.ex [hunk 2]
```diff
@@ -1437,11 +1437,18 @@ defmodule Kernel.SpecialForms do
...
+
+      iex> take_five = &Enum.take(&1, 5)
+      iex> take_five(1..10)
+      [1, 2, 3, 4, 5]
+
+  Another example while using a imported or local function:
```
Comment (josevalim) [inline lib/elixir/lib/kernel/special_forms.ex]: "a imported" => "an imported"

File: lib/elixir/lib/kernel/special_forms.ex (hunk 1 above)
Comment (lexmag) [inline lib/elixir/lib/kernel/special_forms.ex]: I guess the sentence should end with colon?

File: lib/elixir/lib/kernel/special_forms.ex [hunk 3]
```diff
@@ -1437,11 +1437,18 @@ defmodule Kernel.SpecialForms do
...
   In other words, `&(&1 * 2)` is equivalent to `fn x -> x * 2 end`.
-  Another example using a local function:
 
-      iex> fun = &is_atom(&1)
-      iex> fun.(:atom)
-      true
+  We can pritially apply a module function with placeholder.
+
+      iex> take_five = &Enum.take(&1, 5)
+      iex> take_five(1..10)
```
Comment (lexmag) [inline lib/elixir/lib/kernel/special_forms.ex]: Missing dot here: `five.(`.

File: lib/elixir/lib/kernel/special_forms.ex (hunk 1 above)
Comment (taiansu) [inline lib/elixir/lib/kernel/special_forms.ex]: Thanks so much for bringing up the term.

**Comment by taiansu** [comment]
Fixed. Thanks for all the review.

File: lib/elixir/lib/kernel/special_forms.ex [hunk 4]
```diff
@@ -1437,11 +1437,18 @@ defmodule Kernel.SpecialForms do
...
-      true
+  We can partially apply a remote function with placeholder:
+
+      iex> take_five = &Enum.take(&1, 5)
+      iex> take_five.(1..10)
+      [1, 2, 3, 4, 5]
+
+  Another example while using an imported or local function:
+
+      iex> first_elem = &elem(&1, 1)
```
Comment (whatyouhide) [inline lib/elixir/lib/kernel/special_forms.ex]: This doesn't work like in the example: `elem/2` in Elixir is 0-indexed so `first_elem` should be `&elem(&1, 0)`.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**